gen/utils/data/class.go

```diff
@@ -0,0 +1,175 @@
+	PropertiesOptional []string
+	PropertiesReadOnly []string
+	// The raw data from the meta file.
+	// Storing the raw data proactively in case we need to access the data at a later stage.
+	RawMetaData map[string]interface{}
+	// Indicates if the class is required when defined as a child in a parent resource.
+	Required bool
+	// The resource name is the name of the resource in the provider, ex "aci_tenant".
+	ResourceName string
+	// The nested attribute name when part of a parent resource.
+	// ex. "aci_relation_from_bridge_domain_to_netflow_monitor_policy" would translate to "aci_relation_to_netflow_monitor_policy".
+	ResourceNameNested string
+	// The relative name (RN) format of the class, ex "tn-{name}".
+	RnFormat string
+	// The supported APIC versions for the class.
+	// Each version range is separated by a comma, ex "4.2(7f)-4.2(7w),5.2(1g)-".
+	// The first version is the minimum version and the second version is the maximum version.
+	// A dash at the end of a range (ex. 4.2(7f)-) indicates that the class is supported from the first version to the latest version.
+	Versions string
+}
+
+type Child struct {
+	// The name of the child class, ex "fvTenant".
+	// This is used as the key for the map of all classes.
+	ClassName string
+	// When it is a relationship class, this is the class to which the relationship points.
+	PointsToClass string
+}
+
+type ClassDocumentation struct {
+	Description string
+	// List of all child classes which are not included inside the resource but have a separate resource
+	// Used to reference child resource in documentation
+	Children []string
+	// List of notes to be added to the top of the documentation
+	Notes []string
+	// List of warnings to be added to the top of the documentation
+	Warnings []string
+	// List of DN formats
+	DnFormats []string
+}
+
+func NewClass(className string) *Class {
+	genLogger.Trace(fmt.Sprintf("Creating new class struct for class: %s.", className))
+
+	// Splitting the class name into the package and short name.
+	// The package and short names are used for the meta file download, documentation links and lookup in the raw data.
+	var shortName, packageName string
+	genLogger.Trace(fmt.Sprintf("Splitting class name '%s' for name space separation.", className))
+	for index, character := range className {
+		if unicode.IsUpper(character) {
+			shortName = className[index:]
+			packageName = className[:index]
+			break
+		}
+	}
+	if shortName == "" || packageName == "" {
+		genLogger.Fatal(fmt.Sprintf("Failed to split class name '%s' for name space separation.", className))
+	}
+
+	return &Class{
+		ClassName:             className,
+		ClassNameShort:        shortName,
+		ClassNameForFunctions: cases.Title(language.Und, cases.NoLower).String(className),
+		ClassNamePackage:      packageName,
+		Properties:            make(map[string]*Property),
+	}
+}
+
+func (c *Class) LoadMetaFile() {
+	genLogger.Trace(fmt.Sprintf("Loading meta file for class '%s'.", c.ClassName))
```

Comment:
```suggestion
	genLogger.Debug(fmt.Sprintf("Loading meta file for class '%s'.", c.ClassName))
```

Reply:
will change here, but there are more cases where I use trace, should these all be debug?